PWGHF/HFC/Tasks/taskCorrelationDsHadrons.cxx

```diff
@@ -523,15 +522,126 @@ struct HfTaskCorrelationDsHadrons {
+      const auto groupedCandidates = candidates.sliceBy(perCollisionCand, collision.globalIndex());
+
+      // generated candidate loop
+      for (const auto& mcParticle : groupedMcParticles) {
+        if (std::abs(mcParticle.pdgCode()) == Pdg::kDS) {
+          // auto mcCollision = mcParticle.template mcCollision_as<soa::Join<aod::McCollisions, aod::MultsExtraMC>>();
+          // multiplicity = mcCollision.multMCFT0A() + mcCollision.multMCFT0C(); // multFT0M = multFt0A + multFT0C
+          hCandidates->Fill(kCandidateStepMcGenAll, mcParticle.pt(), multiplicityGen, mcParticle.originMcGen());
+          if ((std::abs(mcParticle.flagMcMatchGen()) == 1 << aod::hf_cand_3prong::DecayType::DsToKKPi) && (mcParticle.flagMcDecayChanGen() == decayChannel)) {
+            hCandidates->Fill(kCandidateStepMcGenDsToKKPi, mcParticle.pt(), multiplicityGen, mcParticle.originMcGen());
+            auto yDs = RecoDecay::y(mcParticle.pVector(), o2::constants::physics::MassDS);
+            if (std::abs(yDs) <= yCandGenMax) {
+              hCandidates->Fill(kCandidateStepMcCandInAcceptance, mcParticle.pt(), multiplicityGen, mcParticle.originMcGen());
+              if (mcParticle.originMcGen() == RecoDecay::OriginType::Prompt) {
+                registry.fill(HIST("hPtCandMcGenPrompt"), mcParticle.pt());
+              }
+              if (mcParticle.originMcGen() == RecoDecay::OriginType::NonPrompt) {
+                registry.fill(HIST("hPtCandMcGenNonPrompt"), mcParticle.pt());
+              }
+            }
+            bool isDaughterInAcceptance = true;
+            auto daughters = mcParticle.template daughters_as<CandDsMcGen>();
+            for (const auto& daughter : daughters) {
+              if (daughter.pt() < ptDaughterMin || std::abs(daughter.eta()) > etaTrackMax) {
+                isDaughterInAcceptance = false;
+              }
+            }
+            if (isDaughterInAcceptance) {
+              hCandidates->Fill(kCandidateStepMcDaughtersInAcceptance, mcParticle.pt(), multiplicityGen, mcParticle.originMcGen());
+              registry.fill(HIST("hPtCandMcGenDaughterInAcc"), mcParticle.pt());
+            }
+          }
+        }
+      }
+
+      // reconstructed candidate loop
+      for (const auto& candidate : groupedCandidates) {
+        if (candidate.pt() < ptCandMin || candidate.pt() > ptCandMax) {
+          continue;
+        }
+        std::vector<float> outputMl = {-1., -1., -1.};
+        if (candidate.isSelDsToKKPi() >= selectionFlagDs) {
+          for (unsigned int iclass = 0; iclass < classMl->size(); iclass++) {
+            outputMl[iclass] = candidate.mlProbDsToKKPi()[classMl->at(iclass)];
+          }
+        } else if (candidate.isSelDsToPiKK() >= selectionFlagDs) {
+          for (unsigned int iclass = 0; iclass < classMl->size(); iclass++) {
+            outputMl[iclass] = candidate.mlProbDsToPiKK()[classMl->at(iclass)];
+          }
+        }
```

Comment:
be careful that this would crash if the value of `selectionFlagDs` chosen by the user is `0`, since for candidates that do not pass the topological preselections in the selector no ML output score is computed and so the vector stored in the table is empty

Reply:
Thanks for the comment, for the moment to avoid this I have put a comment on the configurable